### PR TITLE
target_include_directories

### PR DESCRIPTION
**Pull request by bramtayl**
Uses `target_include_directories` instead of `include_directories` for more fine-grain control.

Eventually, it would be good for the plugins and utilities not to use the private include directories, but currently they do.

### CODE REVIEW

File: InOut/CMakeLists.txt [hunk 1]
```diff
@@ -157,25 +155,14 @@ if(USE_IPMIDI)
...
-            ${JACK_LIBRARY} ${PTHREAD_LIBRARY})
-    elseif(WIN32)
-        check_deps(JACK_INCLUDE_PATH)
-        include_directories("${JACK_INCLUDE_PATH}")
-        list(APPEND rtjack_LIBS
-            ${JACK_LIBRARY} ${PTHREAD_LIBRARY})
-    else()
-        list(APPEND rtjack_LIBS
-            ${JACK_LIBRARY} ${PTHREAD_LIBRARY})
-    endif()
```
Comment (stekyne) [inline InOut/CMakeLists.txt:174]: Why did you remove this code?

Comment (bramtayl) [reply InOut/CMakeLists.txt:174]: Well, `list(APPEND rtjack_LIBS ${JACK_LIBRARY} ${PTHREAD_LIBRARY})` was in all 3 branches. And we'd need the jack include directory regardless because we `#include <jack/jack.h>`

File: CMakeLists.txt [hunk 2]
```diff
@@ -568,14 +571,17 @@ endif()
...
 
-include_directories(./H)
-include_directories(./include)
-include_directories(./Engine)
+list(APPEND libcsound_private_include_dirs ${Csound_SOURCE_DIR}/H)
+list(APPEND libcsound_public_include_dirs 
+    $<BUILD_INTERFACE:${Csound_SOURCE_DIR}/include>
+    $<INSTALL_INTERFACE:include>
+)
+list(APPEND libcsound_private_include_dirs ${Csound_SOURCE_DIR}/Engine)
```
Comment (stekyne) [inline CMakeLists.txt:579]: These first and last lines can be combined no?

File: CMakeLists.txt [hunk 3]
```diff
@@ -35,6 +35,9 @@ project(Csound)
 
 enable_testing()
 
+set(libcsound_private_include_dirs "")
+set(libcsound_public_include_dirs "")
```
Comment (stekyne) [inline CMakeLists.txt:39]: Should these be upper case just to be consistent? Could probably just be `LIBCSOUND_PRIVATE_INCLUDES` etc.

**Comment by stekyne** [comment]
This does look like a good change. I think if we can fix the formatting a bit, we can merge this.

EDIT: Just going to merge this now actually seeing as it's relevant to something I'm working on also
